torch/nn/parallel/distributed.py

```diff
@@ -296,7 +296,7 @@ def _process_batch():
             # Wait for all copies to complete before starting the NCCL kernel
             for stream in reduction_streams:
                 stream.synchronize()
-            nccl.reduce(dev_coalesced, root=device_ids[0], streams=nccl_streams)
+            nccl.reduce(dev_coalesced, root=0, streams=nccl_streams)
```

Comment:
This doesn't seem right. We're not guaranteed that 0 is in `device_ids`. Also, it'd be better to completely skip this reduction call if there's only a single GPU. It might also be possible to skip some more calls in `sync_params` in this case.